### PR TITLE
HOCS-5271 move testing to GitHub actions

### PR DESCRIPTION
**Pull request by nickmcmahon01**

### CODE REVIEW

File: ci/localstack/1-setup-queue.sh [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+export AWS_SECRET_ACCESS_KEY=UNSET
+export AWS_DEFAULT_REGION=eu-west-2
+
+## make sure that localstack is running in the pipeline
+until curl http://localstack:4566/health --silent | grep -q "\"sqs\": \"available\""; do
+   sleep 5
+   echo "Waiting for LocalStack to be ready..."
+done
+
+awslocal sqs create-queue --queue-name search-queue-dlq
```
Comment (LiamMacP) [inline ci/localstack/1-setup-queue.sh:14]: This only creates the search queue - it looks like it needs the audit queues to exist to?

File: ci/localstack/3-setup-index.sh [hunk 2]
```diff
@@ -0,0 +1,23 @@
+#!/bin/bash
```
Comment (LiamMacP) [inline ci/localstack/3-setup-index.sh:1]: This isn't required here.

File: ci/localstack/2-setup-elastic.sh [hunk 3]
```diff
@@ -0,0 +1,14 @@
+#!/bin/bash
```
Comment (LiamMacP) [inline ci/localstack/2-setup-elastic.sh:1]: This isn't required here.

File: ci/wait-for-infrastructure.sh [hunk 4]
```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
+set -euo
+
+until curl 'localhost.localstack.cloud:4566/es/eu-west-2/decs/_cat/indices?v' --silent | grep -q "local-case"; do
```
Comment (LiamMacP) [inline ci/wait-for-infrastructure.sh:4]: I do wonder what we can use here instead of ES.

File: docker-compose.yml [hunk 5]
```diff
@@ -1,4 +1,4 @@
-version: '3.8'
+version: '3.9'
```
Comment (LiamMacP) [inline docker-compose.yml:1]: Interesting the docker website doesn't state there is a `3.9`: https://docs.docker.com/compose/compose-file/compose-versioning/

File: docker-compose.yml [hunk 6]
```diff
@@ -16,45 +16,22 @@ services:
...
-        aws --endpoint-url=http://localstack:4576 sqs create-queue --queue-name audit-queue-dlq
-        aws --endpoint-url=http://localstack:4576 sqs create-queue --queue-name search-queue
-        aws --endpoint-url=http://localstack:4576 sqs create-queue --queue-name search-queue-dlq
-        aws --endpoint-url=http://localstack:4575 sns create-topic --name hocs-audit-topic
-        aws --endpoint-url=http://localstack:4575 sns subscribe --topic-arn arn:aws:sns:eu-west-2:123456789012:hocs-audit-topic --protocol sqs --notification-endpoint arn:aws:sns:eu-west-2:123456789012:audit-queue
-        aws --endpoint-url=http://localstack:4575 sns subscribe --topic-arn arn:aws:sns:eu-west-2:123456789012:hocs-audit-topic --protocol sqs --notification-endpoint arn:aws:sns:eu-west-2:123456789012:search-queue
-        aws --endpoint-url=http://localstack:4575 sns list-subscriptions
-        aws --endpoint-url=http://localstack:4576 sqs create-queue --queue-name document-queue
-        aws --endpoint-url=http://localstack:4576 sqs create-queue --queue-name document-queue-dlq
-        aws --endpoint-url=http://localstack:4572 s3 mb s3://untrusted-bucket
```
Comment (LiamMacP) [inline docker-compose.yml:50]: Is this required? I don't see anything to set this up?

File: docker-compose.yml [hunk 7]
```diff
@@ -16,45 +16,22 @@ services:
...
+      - "4566:4566"
     networks:
-      - hocs-network
+      hocs-network:
+        aliases:
+          - localhost.localstack.cloud
     environment:
       HOSTNAME_EXTERNAL: localstack
       DEFAULT_REGION: eu-west-2
-      SERVICES: sns,sqs,s3
```
Comment (LiamMacP) [inline docker-compose.yml:30]: I'd have thought this would be required as it effectively activates them within localstack.

**Review by LiamMacP — CHANGES_REQUESTED** [review]
Looks like there is a few things to still look at 🙂

File: ci/localstack/2-setup-elastic.sh (hunk 3 above)
Comment (nickmcmahon01) [reply ci/localstack/2-setup-elastic.sh:1]: you're right

**Pull request closed without merging**